### PR TITLE
Fixes # 28492 - Use formik instead of redux-forms

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
[test foreman_templates]

**Comment by sharvit** [comment]
Thanks @xprazak2 

IMO the form is too complicated because there is one form that **acts** as 2 different forms and I would like to suggest some refactoring.

1. Split the form into 2 forms. ImportForm and ExportForm.
2. Use tabs instead of radio-buttons and switch them by react-router.

Thoughts?

**Comment by xprazak2** [comment]
Mapping each form to a separate route provided by react-router would be nice, I opened [a new redmine issue](https://projects.theforeman.org/issues/28615) for that and refactor after we swap the form framework.

File: webpack/components/NewTemplateSync/components/NewTemplateSyncForm/__tests__/__snapshots__/NewTemplateSyncForm.test.js.snap [hunk 1]
```diff
@@ -1,11 +1,116 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`NewTemplateSyncForm should render for export settings 1`] = `
-<Form
-  disabled={false}
-  errorTitle="Warning! "
+<ForemanForm
```
Comment (sharvit) [inline webpack/components/NewTemplateSync/components/NewTemplateSyncForm/__tests__/__snapshots__/NewTemplateSyncForm.test.js.snap]: IMO this test/snapshot falls into the definition of a **False positive** and **False negative**.

Most of the data in the snapshot look irrelevant while the only relevant field is the `initialValues`.

File: webpack/components/NewTemplateSync/components/SyncSettingFields.js [hunk 2]
```diff
@@ -58,14 +42,15 @@ const SyncSettingsFields = ({
   return (
     <React.Fragment>
       {settingsAry
-        .map(setting => addValidationToSetting(setting, validationData))
+        .map(addRequiredToSetting)
```
Comment (sharvit) [inline webpack/components/NewTemplateSync/components/SyncSettingFields.js]: The way it works is a bit odd, the `SyncSettingsFields` receives `importSettings`, `exportSettings` and `syncType`.
Based on the `syncType` it will use `importSettings` or `exportSettings`.

IMO it should be done at a higher level so this component would only receive the final `settingsAry`.
Same for the manipulations done on the `settingsAry`, it can all be done in a **memorized selector**.

Please consider refactoring, maybe it should be done in a different PR.

**Comment by xprazak2** [comment]
Those are very good points @sharvit! I'd be definitely willing to improve the structure but I'd prefer to do it in a separate PR to make it obvious which changes were maintenance refactoring and what was needed for formik.

**Comment by sharvit** [comment]
> Those are very good points @sharvit! I'd be definitely willing to improve the structure but I'd prefer to do it in a separate PR to make it obvious which changes were maintenance refactoring and what was needed for formik.

Make sense, thanks.
My only concern is the test snapshots, it looks like things might break but the tests won't catch it (False positive). And from the other side, the test may produce errors while the app would keep functioning (False negative).

False-positive might happen because the snapshot only validates the `initialFields`.
False-negative might happen because the snapshot contains the `validationSchema` that might be changed when we will change/update the validations library.

Is there anything we can do about it before the refactoring?

**Comment by xprazak2** [comment]
[test foreman_templates]

**Comment by xprazak2** [comment]
@ares, can we merge?
